cpu/stm32_common/periph/eeprom.c

```diff
@@ -30,25 +33,155 @@
 
 extern void _lock(void);
 extern void _unlock(void);
+extern void _wait_for_pending_operations(void);
 
 #ifndef EEPROM_START_ADDR
 #error "periph/eeprom: EEPROM_START_ADDR is not defined"
 #endif
 
-uint8_t eeprom_read_byte(uint32_t pos)
+static void _eeprom_write_byte(uint32_t pos, uint8_t byte)
 {
     assert(pos < EEPROM_SIZE);
+    
+    _wait_for_pending_operations();
+    *(__IO uint8_t *)(EEPROM_START_ADDR + pos) = byte;
```

Comment:
`__IO` is not needed I think. But if you have a good reason to keep it, then it should also be used in read functions.